### ISSUE TITLE
Check all variable rescalling in DiVertR21 and make sure it matches what we have in training

### ISSUE THREAD

**Issue by gordonwatts** [gordonwatts/CalRatioTrainer#90]
Or vice versa. But scary.

For example:

Tracking pT is just divided
```c++
                if (track_pt_arr[i] != 0 ) track_pt_arr[i] = (track_pt_arr[i])/(500000);
```
In the training it the min-pt is subtracted, and then scaled by the delta between min and max.

Second, we do a subtraction in our training, but in C++ we don't seem to:
```c++
    inputs["jet_node"] = {{"jet_pt",(jet_pt)/(500000.)}, {"jet_eta",jet_eta}, {"jet_phi", jet_phi}};
    jets->push_back((jet_pt)/(500000.));
```
There is no min-pT subtraction!

And then in the training we have things like teh following:
```c++
    jet_pT_mask = (array.jet_pT >= 40) & (array.jet_pT < 500)
    jet_eta_mask = (array.jet_eta >= -2.5) & (array.jet_eta <= 2.5)
```

So - there is a 500 GeV cut. So what happens to high $p_T$ cuts? Was that ever looked at - NN's unless trained (like the function flow technique) don't always behave. I guess we see good behavior at high pT, so that is ok... But...

And then there is this - no proper phi subtraction!:

```c++
               indexArray[i] = i;
                if (mseg_etaPos_arr[i] != 0)  mseg_etaPos_arr_copy[i] = mseg_etaPos_arr[i] - jet_eta;
                else mseg_etaPos_arr_copy[i] = mseg_etaPos_arr[i];
                if ( mseg_phiPos_arr[i] != 0 ) mseg_phiPos_arr[i] = mseg_phiPos_arr[i] - jet_phi;
                else mseg_phiPos_arr[i] = 9999999;
                mseg_etaDir_arr_copy[i] = mseg_etaDir_arr[i];
                if ( mseg_phiDir_arr[i] != 0) mseg_phiDir_arr_copy[i] = mseg_phiDir_arr[i] - jet_phi;
                else mseg_phiDir_arr_copy[i] = mseg_phiDir_arr[i];
                mseg_t0_arr_copy[i] = mseg_t0_arr[i];
                mseg_chiSquared_arr_copy[i] = mseg_chiSquared_arr[i];
```

**Comment by gordonwatts** [comment]
Here is some more odd evidence. Here are plots from the paper of the rescaled jet $p_T$:

![image](https://github.com/gordonwatts/CalRatioTrainer/assets/1778366/2ab874e9-944c-4983-8ac8-432f11817fca)

Note that it doesn't go down to zero - so that means the training data probably was actually rescaled the same way as the data in the program - and not how it was described in the paper!! In short, the min jet pT is 0 when talking about the rescaling, but the cut is in there.

Which also makes little sense because in the code the cut and the rescaling are the same variable!

**Comment by gordonwatts** [comment]
Here is a table of everything that is being done. In C++ padding is always done with zero's.

| Variable | C++ | CalRatioTrainer | Notes |
| --- | --- | --- | --- |
| jet pt | Divided by 500.  |  | Units are GeV  |
| jet $\eta$ | If not zero, adjusted so first cluster is at zero | | |
| jet $\phi$ | If not zero, adjusted so first cluster is at zero | | |
| Cluster pt| Divided by 500.0, sorted by pT from largest to smallest. Conversion to GeV is done in an odd place. | | Units are GeV |
| Cluster eta| if the $p_T$ weighted sum of all $\eta < 0.0$, then flip the sign of the eta | | |
| Cluster Energies | Each made to be a fraction of the sum of all cluster energies | | These include the presampler as the first layer, and combine the forward and barrel layers |
| track pt | Divide by 500, sorted from largest to smallest | | Only tracks with a $\Delta R < 0.2$ are included. Units are GeV. |
| track eta| If not zero, adjusted so jet $\eta$ is zero, if eta sum is less than zero then flipped | | TODO: Why is this done differently from the cluster eta flip? Shouldn't the same flip happen in one place or the other? |
| track phi| If not zero, adjusted so jet $\phi$ is zero | | |
| mseg $\eta$ position | If not zero, adjusted so jet $\eta$ is zero | | Only mseg with position $\Delta R < 0.2$ are included. These are not sorted due to instability of sort. |
| mseg $\phi$ position | If not zero, adjusted so jet $\phi$ is zero | | |
| mseg $\phi$ direction | if not zero, adjusted so jet $\phi$ is zero | | |

**Comment by gordonwatts** [comment]
Added the changed 2019 tag as it gives a nice listing of all the preprocessing that goes on pre-2019 as well as now. There could be one more column - what is written up in our internal note. :-)
